### ISSUE TITLE
Not adding dynamic components to dependencies

### ISSUE THREAD

**Issue by adrianhelvik** [yuchi/hooks.macro#16]
Found an issue with detecting variable usage when the components are stored as variables. We use this quite frequently in our app.

**Repro**
```javascript
import { useAutoMemo } from 'hooks.macro'
import React from 'react'

const Foo = () => null
const Bar = () => null

export default function MyComponent({ bar }) {
  const Selected = Foo

  if (foo) Selected = Bar

  return useAutoMemo(<Selected />)
}
```

**Output:**
```javascript
import { useMemo as _useMemo } from "react";
import React from 'react';

const Foo = () => null;

const Bar = () => null;

export default function MyComponent({
  bar
}) {
  const Selected = Foo;
  if (bar) Selected = Bar;
  return _useMemo(() => /*#__PURE__*/React.createElement(Selected, null), []);
}
```
**Expected:**
```javascript
import { useMemo as _useMemo } from "react";
import React from 'react';

const Foo = () => null;

const Bar = () => null;

export default function MyComponent({
  bar
}) {
  const Selected = Foo;
  if (bar) Selected = Bar;
  return _useMemo(() => /*#__PURE__*/React.createElement(Selected, null), [Selected]);
}
```

**Comment by adrianhelvik** [comment]
The test for do-expressions can on the same note be fixed by adding:
```javascript
      babelOptions: {
        babelrc: true,
        filename: path.join(__dirname, 'do-expressions.js'),
      },
```
to the test case.

**Comment by yuchi** [comment]
This is indeed a very nasty bug — we actually missed a reference there! Thank you so much for reporting this.

I found a slighlty more succint solution, different from your approach, but will add you as a co-editor since you gave most of the details right.

**Comment by yuchi** [comment]
Fixed in `v1.1.2`! 🎉 Thank you so much!
